### PR TITLE
Lookup join on multiple join fields not yet supported

### PR DESCRIPTION
**Pull request by astefan**
`lookup join`s shouldn't work on multiple join fields at the moment.
This PR also adds tests for another existent limitation: joining fields must be only actual index fields (not constants for example).

### CODE REVIEW

**Review by bpintea — APPROVED** [review]
Left a note, LG otherwise.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java [hunk 1]
```diff
@@ -564,6 +564,11 @@ public PlanFactory visitJoinCommand(EsqlBaseParser.JoinCommandContext ctx) {
             }
         }
 
+        var matchFieldsCount = joinFields.size();
+        if (matchFieldsCount > 1) {
```
[the remark this answers is not in the document]
Comment (astefan) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java:568]: Do you mean `lookup join test on languages, languages` should be considered as `lookup join test on languages`?

[the remark this answers is not in the document]
Comment (bpintea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java:568]: Yes, that was what I was thinking. Or, when we might support more, `ON x, y, x` be same as `ON x, y`.
But can be left as is as well.

[the remark this answers is not in the document]
Comment (astefan) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java:568]: Imo, we are hiding a behavior (eliminating duplicates) without the user being notified of, what is in essence, an incorrect query. I _think_ for now I will leave this as is in this PR and relax things later, if we identify this as a drawback. Thanks.

**Pull request merged**